hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestNestedSchemaPruningOptimization.scala

```diff
@@ -113,6 +113,9 @@ class TestNestedSchemaPruningOptimization extends HoodieSparkSqlTestBase with Sp
               //assertEquals(tableName, tableIdentifier.get.table)
               //assertEquals(expectedSchema, requiredSchema, hint)
           }
+
+          // Execute the query to make sure it's working as expected (smoke test)
+          selectDF.count
```

Comment:
Did this fail due to incorrect nested schema pruning before this PR?

Reply:
Correct